Services/Services.Application/Services/ServiceCategoryManager.cs

```diff
@@ -0,0 +1,72 @@
+﻿using AutoMapper;
+using Services.Application.Models;
+using Services.Domain.Repositories.Abstractions;
+using Services.Application.Services.Abstractions;
+using Services.Domain.Entities;
+using Services.Application.Exceptions;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Services.Application.Services
+{
+    public class ServiceCategoryManager : IServiceCategoryManager
+    {
+        private readonly IUnitOfWork _unitOfWork;
+        private readonly IMapper _mapper;
+
+        public ServiceCategoryManager(IUnitOfWork unitOfWork, IMapper mapper)
+        {
+            _mapper = mapper;
+            _unitOfWork = unitOfWork;
+        }
+
+        public async Task Create(CreateServiceCategoryModel createModel, CancellationToken cancellationToken)
+        {
+            await _unitOfWork.ServiceCategoryRepository.CreateAsync(_mapper.Map<ServiceCategory>(createModel), cancellationToken);
+            await _unitOfWork.SaveChangesAsync(cancellationToken);
+        }
+
+        public async Task Delete(Guid id, CancellationToken cancellationToken)
+        {
+            var servicesRelatedToCategory = await _unitOfWork.ServiceRepository.GetActiveServicesByCategoryIdAsync(id, cancellationToken);
```

Comment:
Can you do one call to db for check those entities? Fix in all places where it possible

Reply:
Fixed